src/rustup.rs

```diff
@@ -104,10 +104,18 @@ pub fn available_targets(toolchain: &str, msg_info: &mut MessageInfo) -> Result<
     })
 }
 
+pub fn install_toolchain_command(toolchain: &str, msg_info: &mut MessageInfo) -> Command {
+    let mut command = rustup_command(msg_info, false);
+    command.args(&["toolchain", "add", toolchain, "--profile", "minimal"]);
+
+    command
+}
+
 pub fn install_toolchain(toolchain: &str, msg_info: &mut MessageInfo) -> Result<()> {
-    rustup_command(msg_info, false)
-        .args(&["toolchain", "add", toolchain, "--profile", "minimal"])
+    install_toolchain_command(toolchain, msg_info)
+        .arg("--force-non-host")
         .run(msg_info, false)
+        .or_else(|_| install_toolchain_command(toolchain, msg_info).run(msg_info, false))
         .wrap_err_with(|| format!("couldn't install toolchain `{toolchain}`"))
 }
```

Comment:
now that I think about this some more, maybe it's not a good idea to have it this simplistic, two rustup calls is probably the best solution. I should've requested/implemented `--force-non-host` as a env var as well :/

Reply:
The reason I think this is not good is because there'll be some strange behaviour if both calls are invalid/rustup is not properly configured.

Reply:
Is there a better solution though? I think the version is better since we capture `stdout` and error if it fails, and then can check if the version is correct. We get the appropriate error message. The downside is we have 2 rustup calls always.

Reply:
Just a ping on this: what's the best design here?

Reply:
lets keep it as is, but don't throw away the first error